src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -1122,7 +1122,8 @@ private void WriteRedType(TreeType node)
                             ? $"GetRedAtZero(ref this.{CamelCase(field.Name)})"
                             : $"GetRed(ref this.{CamelCase(field.Name)}, {index})";
 
-                        WriteLine($" => index == {index} ? {whenTrue} : null;");
+                        var suffix = IsOptional(field) ? "" : "!";
```

Comment:
It's unfortunate that we don't have sequence expressions, because I would very much like to assert that these are NotNull. Perhaps we should add a method like this:
```cs
[return: NotNull]
public T AssertNotNull<T>(T t)
{
    RoslynDebug.Assert(t is object);
    return t;
}
```
and then call that here, rather than suppressing warnings?

Reply:
And use it in the other places we're using suppressions currently.

Reply:
Note: this is orthogonal to this review.  This maintains the work that Sam did.  I really don't want to generate anything *different* at runtime versus what we're generating now.  That's out of scope for this PR.

Reply:
Alright, that's fine.